### ISSUE TITLE
CMake install target for headers is incorrect + Quick fix

### ISSUE THREAD

**Issue by NaoPross** [ThePhD/sol2#831]
Thanks for your amazing work! Though I need to report a quick problem I found in `CMakeLists.txt`.
The headers that should be installed under `${CMAKE_INSTALL_INCLUDEDIR}/include/sol` are instead installed in a redundant subfolder `${CMAKE...}/include/sol/include/sol`.

I'm using CMake 3.10 under Linux (Ubuntu derivative) and Ninja to build.
```
naopross@saturn:build$ sudo ninja install
[0/1] Install the project...
-- Install configuration: ""
-- Installing: /usr/local/lib/cmake/sol2/sol2-targets.cmake
-- Installing: /usr/local/include/sol/include
-- Installing: /usr/local/include/sol/include/sol
-- Installing: /usr/local/include/sol/include/sol/function_types_core.hpp
-- Installing: /usr/local/include/sol/include/sol/stack_check_get_qualified.hpp
...
```
I'm not a CMake guru but here's a quick patch that fixed it, based off commit `dd95030b`. Or let me know if I'm using CMake the wrong way.
```patch
From c087a0bdd84c9a96ef710de8f6f2c27fc7c0eae0 Mon Sep 17 00:00:00 2001
From: Nao Pross <naopross@thearcway.org>
Date: Mon, 24 Jun 2019 17:46:43 +0200
Subject: [PATCH] Fix CMake headers install target

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 961b39d6..698754cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,7 +113,7 @@ set_target_properties(sol2
 
 target_include_directories(sol2 INTERFACE
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-	$<INSTALL_INTERFACE:sol/include>)
+	$<INSTALL_INTERFACE:include>)
 
 # # Version configurations
 configure_package_config_file(
@@ -135,8 +135,8 @@ install(EXPORT sol2
 	FILE sol2-targets.cmake
 	DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/sol2")
 
-install(DIRECTORY include/
-	DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/sol/include")
+install(DIRECTORY include/sol/
+	DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/sol/")
 
 install(FILES
 	"${CMAKE_CURRENT_BINARY_DIR}/cmake/sol2-config.cmake"
-- 
2.17.1
```
[0001-Fix-CMake-headers-install-target.patch.txt](https://github.com/ThePhD/sol2/files/3321672/0001-Fix-CMake-headers-install-target.patch.txt)
(Appy with `$ git am < 0001-Fix-CMake-headers-install-target.patch.txt`)

**Comment by ThePhD** [comment]
Thanks, I'll apply the patch.

FWIW I have no idea what I'm doing with the install targets so literally none of what I've written is intentional.

**Comment by NaoPross** [comment]
Hey, I have open this again because you patched wrong (why not use `git am` or `patch -p1`?).

See: https://github.com/ThePhD/sol2/commit/da789b067c06a15afa67d4fbbae94b053c2812fc#r34133022

**Comment by ThePhD** [comment]
I'm confused. Is it supposed to be

`${CMAKE_INSTALL_INCLUDEDIR}/sol/`

or `${CMAKE_INSTALL_INCLUDEDIR}` ?

The goal is to have it be under `include/sol/sol.hpp`

**Comment by NaoPross** [comment]
Okay, I've quickly tested both cases. N.B. I am using the normal `sol2::sol2` target **not**  `sol2::sol2_single`, later I may check if everything works with `sol2_single`.

I don't know if it is Linux specific or general under *nix systems but there is a difference between for example `dir` and `dir/`, because the first is "the directory `dir`" whereas the latter means "the content of `dir`". Which makes things complicated, and gives you many options:

You either use
```CMake
install(DIRECTORY include/sol/
	DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/sol")
```
Which reads: the content of `include/sol` goes inside the directory `${CMAKE...}/sol`. (correct)

Or
```CMake
install(DIRECTORY include/
	DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
```
The content of `include` (i.e. the `sol` directory) goes inside `${CMAKE_INSTALL_INCLUDEDIR}` (correct)

Or
```CMake
install(DIRECTORY include/sol
	DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
```
The directory `include/sol` goes inside `${CMAKE_INSTALL_INCLUDEDIR}` (also correct)

Though I tested a few combinations and it seems like this trailing `/` business applies only to the `DIRECTORY` argument. I expected this to work but it doesn't:
```CMake
install(DIRECTORY include/sol/
	DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/sol/")
```
From which I expected: the content of `include/sol` goes into `${CMAKE...}/sol`. (wrong!)

This last one does something else and puts `include/sol` under `${CMAKE...}/sol` creating `${CMAKE...}/sol/sol/`. Why? No clue, I'll never understand the garbled mess that is CMake.


TLDR: any of the above **except the last one** work, you can choose the one you want (because there could be other stuff inside your `include` dir).

**Comment by ThePhD** [comment]
I picked one, it should work.
